### PR TITLE
Fix new clippy lints

### PR DESCRIPTION
**Pull request by GuillaumeGomez**

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
New warning is wild:

```
error: use of deprecated constant `libc::MSG_OOB`: recvmmsg call expects an unsigned type on musl/emscripten
  --> gio/sys/src/manual.rs:32:64
   |
32 |     pub const G_SOCKET_MSG_OOB: super::GSocketMsgFlags = libc::MSG_OOB;
   |                                                                ^^^^^^^
   |
   = note: `-D deprecated` implied by `-D warnings`
```

Any idea what's going on?

**Comment by AaronErhardt** [comment]
@GuillaumeGomez looks like a bunch of glibc types were deprecated with no real alternative in place yet (so this has nothing to do with clippy actually). It seems like this got reverted already: https://github.com/rust-lang/libc/pull/2978

**Comment by GuillaumeGomez** [comment]
Thanks for the information @AaronErhardt !

**Comment by sdroege** [comment]
Yeah, let's not worry about those for now.

File: gio/src/lib.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 #![allow(clippy::missing_safety_doc)]
 #![allow(clippy::upper_case_acronyms)]
 #![allow(clippy::non_send_fields_in_send_ty)]
+#![allow(clippy::should_implement_trait)]
```
Comment (sdroege) [inline gio/src/lib.rs:10]: This is generally a good lint. Can you only allow it for the one instance where it's wrong?

File: gio/src/list_model.rs [hunk 2]
```diff
@@ -103,8 +103,8 @@ impl<'a, T: IsA<glib::Object>> Iterator for ListModelIter<'a, T> {
     }
 
     fn size_hint(&self) -> (usize, Option<usize>) {
-        let n = (self.reverse_pos - self.i) as usize;
-        (n as usize, Some(n))
+        let n: usize = (self.reverse_pos - self.i) as _;
+        (n as _, Some(n))
```
Comment (sdroege) [inline gio/src/list_model.rs:107]:
```suggestion
        (n, Some(n))
```

You already make it an `usize` before. This warning was actually correct

File: glib/src/lib.rs [hunk 3]
```diff
@@ -5,6 +5,7 @@
 #![allow(clippy::missing_safety_doc)]
 #![allow(clippy::wrong_self_convention)]
 #![allow(clippy::non_send_fields_in_send_ty)]
+#![allow(clippy::should_implement_trait)]
```
Comment (sdroege) [inline glib/src/lib.rs:8]: also here

File: pangocairo/src/lib.rs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
+#![allow(clippy::should_implement_trait)]
```
Comment (sdroege) [inline pangocairo/src/lib.rs:3]: Also here

File: gio/src/lib.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply gio/src/lib.rs:10]: That would require another gir update but I can do that.

Comment (sdroege) [reply gio/src/lib.rs:10]: As discussed with @bilelmoussaoui, just put it on the `auto` module for now via `src/lib.rs`

File: pangocairo/src/lib.rs (hunk 4 above)
Comment (GuillaumeGomez) [reply pangocairo/src/lib.rs:3]: Ah too late. I put the attribute on the methods directly.

File: gio/src/list_model.rs (hunk 2 above)
Comment (sdroege) [reply gio/src/list_model.rs:107]: You didn't change anything here :)

Comment (GuillaumeGomez) [reply gio/src/list_model.rs:107]: I did one `git checkout --` too much.

**Comment by GuillaumeGomez** [comment]
Updated!

**Review by sdroege — APPROVED** [review]

**Pull request merged**